Source/Core/Core/State.cpp

```diff
@@ -431,14 +431,11 @@ void SaveAs(const std::string& filename, bool wait)
         {
           Core::DisplayMessage("Saving State...", 1000);
 
-          CompressAndDumpState_args save_args;
-          save_args.buffer_vector = &g_current_buffer;
-          save_args.buffer_mutex = &g_cs_current_buffer;
-          save_args.filename = filename;
-          save_args.wait = wait;
+          CompressAndDumpState_args save_args = {&g_current_buffer, &g_cs_current_buffer, filename,
```

Comment:
Perhaps leave a todo comment here to turn this into designated initializers when moving over to C++20 in the future